lib/properties.ts

```diff
@@ -171,18 +176,50 @@ export class CompilerProps {
      * @returns {*} Transformed value(s) found or fn(defaultValue)
      */
     get(
-        langs: string | LanguageDef[],
+        base: string | LanguageDef[] | Record<string, any>,
+        property: string,
+        defaultValue?: undefined,
+        fn?: undefined,
+    ): PropertyValue;
+    get<T extends PropertyValue>(
+        base: string | LanguageDef[] | Record<string, any>,
+        property: string,
+        defaultValue: Widen<T>,
+        fn?: undefined,
+    ): typeof defaultValue;
+    get<T extends PropertyValue>(
+        base: string | LanguageDef[] | Record<string, any>,
+        property: string,
+        defaultValue?: PropertyValue,
+        fn?: undefined,
+    ): T;
+
+    get<R>(
+        base: string | LanguageDef[] | Record<string, any>,
+        property: string,
+        defaultValue?: undefined,
+        fn?: (item: PropertyValue, language?: any) => R,
+    ): R;
+    get<T extends PropertyValue, R>(
+        base: string | LanguageDef[] | Record<string, any>,
+        property: string,
+        defaultValue: Widen<T>,
+        fn?: (item: PropertyValue, language?: any) => R,
+    ): typeof defaultValue | R;
+
+    get(
+        langs: string | LanguageDef[] | Record<string, any>,
         key: string,
-        defaultValue: PropertyValue,
-        fn: (item: PropertyValue, language?: any) => PropertyValue = _.identity,
+        defaultValue?: PropertyValue,
+        fn?: (item: PropertyValue, language?: any) => unknown,
     ) {
         fn = fn || _.identity;
         if (_.isEmpty(langs)) {
             return fn(this.ceProps(key, defaultValue));
         }
         if (!_.isString(langs)) {
             return _.chain(langs)
-                .map(lang => [lang.id, fn(this.$getInternal(lang.id, key, defaultValue), lang)])
+                .map(lang => [lang.id, fn!(this.$getInternal(lang.id, key, defaultValue), lang)])
```

Comment:
Wait why does it not recognize that it can't be undefined at this point anymore? I'd have thought that line 216 would narrow the type enough

Reply:
This puzzled me as well! It's because it's in an arrow function and I guess they don't want to special-case `.map`